CRM/Member/BAO/Membership.php

```diff
@@ -1915,6 +1915,9 @@ public static function createRelatedMemberships(&$params, &$dao, $reset = FALSE)
         unset($params['lineItems']);
         unset($params['line_item']);
 
+        // CRM-20966: Do not create membership_payment record for inherited membership.
+        unset($params['relate_contribution_id']);
+
```

Comment:
@KarinG Thanks for looking at this. Good question to ask. As I understand it, relate_contribution_id is one of the params set for creating the primary membership, in CRM_Member_Form_Membership::postProcess() . These params then get used as the basis for creating the related memberships, with some meddling first. Other params get removed before the related membership is created, e.g. id, reminder_date, custom data, max_related, contribution_status_id, lineItems/line_item. So this seemed a reasonable place to remove it.